closed/src/java.base/share/native/libjncrypto/NativeCrypto.c

```diff
@@ -38,10 +39,15 @@
 #include "jdk_crypto_jniprovider_NativeCrypto.h"
 #include "NativeCrypto_md.h"
 
-#define OPENSSL_VERSION_1_0 "OpenSSL 1.0."
-#define OPENSSL_VERSION_1_1 "OpenSSL 1.1."
+#define OPENSSL_VERSION_CODE(major, minor, fix, patch) \
+        (((major) << 28) | ((minor) << 20) | ((fix) << 12) | (patch))
```

Comment:
Suggest adding a cast to `jlong` here to avoid loss of information when `major` is 16 or more:
```c
#define OPENSSL_VERSION_CODE(major, minor, fix, patch) \
        ((((jlong)(major)) << 28) | ((minor) << 20) | ((fix) << 12) | (patch))
```

Reply:
Updated.